packages/rum-core/src/common/apm-server.js

```diff
@@ -150,7 +155,20 @@ class ApmServer {
         const { status, responseText } = xhr
         reject({ url, status, responseText })
       }
-      xhr.send(payload)
+
+      let canSend = true
+      if (typeof beforeSend === 'function') {
+        canSend = beforeSend({ url, method, headers, payload, xhr })
+      }
+      if (canSend) {
+        xhr.send(payload)
+      } else {
+        reject({
```

Comment:
forgot to catch this before, do we want to fail here explicitly which will show up as warnings in console `Failed sending events` or silently fail by resolve it with empty response text?

Reply:
Yeah, the warning is useful if users have made a mistake in their callback function.